### PR TITLE
Add new storage quota exception for geocoding and remove premature fail status

### LINKED ISSUE #5230: Add geocoding error message for errors caused by excess of quota
When performing a geocoding process, adding the geocoded information to the table might exceed the quota set up for the user. In these cases, we get internally errors like:

`Sequel::DatabaseError: PG::Error: ERROR: Quota exceeded by 1460KB`

but in the UI we're just showing a generic Geocoding Error.

What about trying to move this inner error to the UI? This will reduce support and will allow the users to understand what's happening.

cc @saleiva

### PR DESCRIPTION
**Pull request by iriberri**
Closes #5230 and helps with https://github.com/CartoDB/cartodb/issues/7338

While working on adding an exception for quota issues, I realised about a problem in the current flow.

Whenever an exception was raised, the status was automatically set to failed. The polling of the frontend was detecting the failed status and setting the geocoding as failed, but with an unknown error because the error is detected afterwards by `handle_geocoding_failure`. The current `handle_geocoding_failure` changes the status to failed (again), so I removed the initial failure setup, which is hiding the real exceptions.

The flow was something like:

* Editor starts geocoding process
* Geocoding fails
* Status is set to failure in the backend
* The Editor detects the new failure state and shows a modal about the error
* The backend keeps working on handling the failure and sets the error code accordingly, which is not shown to the user

### CODE REVIEW

**Comment by iriberri** [comment]
Hi @ethervoid, you were the last one working with this part of the code -- specially with the "set status to failure" line that I'm attempting to remove. Could you please take a look? :-)

File: services/table-geocoder/lib/internal_geocoder.rb [hunk 1]
```diff
@@ -133,6 +132,10 @@ def copy_results_to_table
             @query_generator.copy_results_to_table_query,
             @table_schema, @table_name
           )
+      rescue => e
```
Comment (ethervoid) [inline services/table-geocoder/lib/internal_geocoder.rb]: You have to mark the result as failed in any moment isn't it?

Comment (iriberri) [inline services/table-geocoder/lib/internal_geocoder.rb]: But it already does here, no? https://github.com/CartoDB/cartodb/blob/master/app/models/geocoding.rb#L370

Comment (ethervoid) [inline services/table-geocoder/lib/internal_geocoder.rb]: Hum, you're right

Comment (iriberri) [inline services/table-geocoder/lib/internal_geocoder.rb]: So you like it? :P

**Comment by ethervoid** [comment]
Sounds good to me but tests it thoroughly because the status part is a tricky one

**Comment by iriberri** [comment]
retest this please

**Comment by iriberri** [comment]
This is gonna need some frontend work, as the HTML in the exceptions is not being interpreted.

**Comment by iriberri** [comment]
I'm gonna close this without deleting the branch for the moment, because I don't think is worthy to spend any more time in the old geocoding process right now.

**Pull request closed without merging**